### ISSUE TITLE
крашит приложение в IE11

### ISSUE THREAD

**Issue by yoshkin** [vitalybaev/react-dadata#87]
Пробовал разные версии от 2.12 до 2.19
Приложение падает с ошибкой "Objects are not valid as a React child..."

Добавление список модулей для лоадера, не решает проблему:
`test: /\.(ts|tsx|js|jsx)$/,
                          include: /node_modules[\\,/](query-string|yup|strict-uri-encode|split-on-first[\\,/]).*/,
                          use: [
                              {
                                  loader: 'ts-loader',
                                  options: {},
                              },
                          ],`

@babel/polyfil также установлен.
Что может быть не так?
Если удалить пакет dadata из сборки - все работает (кроме подсказок)

**Comment by yoshkin** [comment]
Да, в других браузерах проблемы нет.

Используется только 1 компонент из пакета и только в 1 месте, вот так:

```
                               <AddressSuggestions
                                    token={DADATA_TOKEN}
                                    delay={400}
                                    count={7}
                                    inputProps={{
                                        value: fields.customDeliveryAddressDHL,
                                        id: 'customDeliveryAddressDHL',
                                        name: 'customDeliveryAddressDHL',
                                        onChange: handleBlurInputCustomAddressDHL,
                                        disabled: !isEditableEnquiry,
                                        placeholder: localization.DELIVERY.INPUT_ADDRESS_PLACEHOLDER,
                                        onBlur: handleBlurInputCustomAddressDHL,
                                        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
                                        // @ts-ignore
                                        invalid: Boolean(
                                            deliveriesErrors.customDeliveryAddressDHL &&
                                                !customAddresses.network.isError
                                        ),
                                        postfix: loadingCustomAddresses,
                                    }}
                                    filterFromBound='city'
                                    onChange={handleChangeCustomAddress}
                                    customInput={Input}
                                />
```

**Comment by vitalybaev** [comment]
Спасибо за информацию! Действительно, кажется, какой-то синтаксис JS приводит к такому. Возможно, надо что-то поменять в транспиляции. Постараюсь поизучать проблему, только вот IE11 под рукой сейчас нет, попробую воспользоваться BrowserStack.

**Comment by yoshkin** [comment]
Да я сам на маке сижу, приходится через виртуалку на удаленном рабочем столе тестировать, тк у некоторых пользователей такая проблема вылезла

**Comment by yoshkin** [comment]
Даже когда оставил только вот такой простой компонент:
```
<AddressSuggestions
  token={DADATA_TOKEN}
  delay={400}
  count={7}
  filterFromBound='city'
  onChange={handleChangeCustomAddress}
/>
```
Все равно проблема осталась в IE11, видимо проблема где-то глубже, не только там, где пропсы прокидываются.

**Comment by yoshkin** [comment]
Пока не было времени проверить? я просто еще поковырялся и не нашел в каком месте пакет крашит IE, из-за особенностей приложения приходиться тестировать на собранном чанке для прода, локально нет возможности в IE проверить

**Comment by vitalybaev** [comment]
К сожалению, пока нет. У меня еще добавилось бытовых дел по личным причинам, но я помню про ишью и постараюсь посмотреть на этих выходных

**Comment by yoshkin** [comment]
Спасибо большое

**Comment by vitalybaev** [comment]
@yoshkin а можно пример репозитория/codesandbox чтобы я смог запустить?

По итогу:

Стартер vite вообще в ie11 не заработал без ошибок.
CRA сначала ругнулся на отсутствия crypto, это из-за nanoid, решается просто

```js
if (typeof window !== 'undefined' && !window.crypto) {
  window.crypto = window.msCrypto;
}
```

А потом начал ругаться на отсутствие reduce у объекта но ошибка из недров реакта. Поэтому было бы здорово, если получится предоставить репозиторий с нужной настройкой и полифилами. Просто я вообще уже хотел дропать поддержку IE 11, как по мне он уже не стоит того времени, чтобы с ним разбираться

Во всех случаях использовал дополнительно `react-app-polyfill`.

**Comment by yoshkin** [comment]
https://codesandbox.io/embed/agitated-booth-2lmpl0?fontsize=14&hidenavigation=1&theme=dark
ключ тут вставил из вашего же примера
Полифилы в моем случае используются из пакета `"@babel/polyfill": "^7.10.1",`

**Comment by yoshkin** [comment]
Может быть как вариант, прямо в самом пакете возвращать условно "false" при инициализации пакета?
А то фронт сейчас крашится просто из-за того, что компонент импортирован в другом компоненте, соответственно, проверки на то, с какого браузера зашел пользователь - в конечном компоненте ни к чему не приведут.

**Comment by vitalybaev** [comment]
Так, codesandbox плохой выбор, потому что у меня в IE на запускается вообще ничего потому что там Codesanbox инджектит свой код, который не работает в IE11.

Причем даже несмотря на полифилл который я в сборку добавляю.

> Может быть как вариант, прямо в самом пакете возвращать условно "false" при инициализации пакета?

Вот тут не понял

**Comment by Eramen** [comment]
Всем привет, возникла ошибка с reduce, о которой писали чуть выше. В ходе исследования было найдено, что узким местом является nanoid, про который в принципе и написано в ошибке при генерации _uid-а в компоненте BaseSuggestion. Это связано с тем, что в ie11 не работает создание массива Uint8Array в том виде, в котором оно сделано в nanoid.

Решается это добавлением polyfill-а 'core-js/es/typed-array'. Может кому пригодится в случае возникновения такой ошибки.

**Comment by vitalybaev** [comment]
Да, я в своё время добавлял в readme nanoid эту особенность. Сейчас доступно в ветке 3-й версии https://github.com/ai/nanoid/tree/v3#ie.

Если проблема была только в этом, то рад, что решилось.

Ишью закрою
